### PR TITLE
Convert HTML header field names to lowercase for HTTP2

### PR DESCRIPTION
**Pull request by cclauss**
HTTP2 requires that header field names be lowercase ss discussed at
https://datatracker.ietf.org/doc/html/rfc7540#section-8.1.2 and https://github.com/internetarchive/openlibrary/pull/5789#issuecomment-963611134

### CODE REVIEW

File: openlibrary/plugins/importapi/code.py [hunk 1]
```diff
@@ -458,12 +458,19 @@ def POST(self):
...
+        return web.HTTPError("400 Bad Request", {"content-type": "application/json"}, d)
 
 
     def auth_failed(self, reason):
         d = json.dumps({ "status" : "error", "reason" : reason})
-        return web.HTTPError("401 Authorization Required", {"WWW-Authenticate": 'Basic realm="http://openlibrary.org"', "Content-Type": "application/json"}, d)
+        return web.HTTPError(
+            "401 Authorization Required",
+            {
+                "www-authenticate": 'Basic realm="http://openlibrary.org"',
```
Comment (mekarpeles) [inline openlibrary/plugins/importapi/code.py:469]: In some places above we've changed `Basic` -> `basic`

There are several other locations here where we don't make that change

https://github.com/internetarchive/openlibrary/pull/5840/files#diff-fefad9036d5577fbc4e634459d3638bae0e47a93269b28302f90c660b276cc7bL37

File: openlibrary/core/civicrm.py [hunk 2]
```diff
@@ -34,7 +34,7 @@ def get_contact(username=None, contact_id=None):
             params=data,
             timeout=3,
             headers=dict(
-                Authorization=f"Basic {lending.config_ia_civicrm_api.get('auth', '')}"
+                Authorization=f"basic {lending.config_ia_civicrm_api.get('auth', '')}"
```
Comment (cclauss) [inline openlibrary/core/civicrm.py:37]: Lowercase keys.  Do not change the values.
```suggestion
                Authorization=f"Basic {lending.config_ia_civicrm_api.get('auth', '')}"
```

File: openlibrary/core/civicrm.py [hunk 3]
```diff
@@ -83,7 +83,7 @@ def get_sponsorships_by_contact_id(contact_id=None, isbn=None):
             timeout=3,
             params=data,
             headers=dict(
-                Authorization=f"Basic {lending.config_ia_civicrm_api.get('auth', '')}"
+                Authorization=f"basic {lending.config_ia_civicrm_api.get('auth', '')}"
```
Comment (cclauss) [inline openlibrary/core/civicrm.py:86]:
```suggestion
                Authorization=f"Basic {lending.config_ia_civicrm_api.get('auth', '')}"
```

File: openlibrary/plugins/importapi/code.py (hunk 1 above)
Comment (cclauss) [reply openlibrary/plugins/importapi/code.py:469]: Nice catch!  Thanks.

File: openlibrary/i18n/cs/messages.po [hunk 4]
```diff
@@ -14,7 +14,7 @@ msgstr ""
 "Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
 "Language-Team: cs <LL@li.org>\n"
 "MIME-Version: 1.0\n"
-"Content-Type: text/plain; charset=utf-8\n"
+"content-type: text/plain; charset=utf-8\n"
```
Comment (jimchamp) [inline openlibrary/i18n/cs/messages.po:17]: I don't think that `Content-Type` should be changed in any of the `.po` files.  This header entry is used by `gettext`, and I can't find any guidance regarding case sensitivity of header entries.  Better safe than sorry.

**Comment by cdrini** [comment]
I'm of the mind we should close this ; it's a pretty big potentially risky change to fix an issue that we haven't observed on the site. This sort of PR is actually exactly what I was trying to avoid by posting https://github.com/internetarchive/openlibrary/issues/5418#issuecomment-963553549 and https://github.com/internetarchive/openlibrary/pull/5789#issuecomment-963647458 . I imagine case sensitivity is handled at a lower level by `requests` or `web.py` , which is why we haven't seen any issues. Changing all these strings in a semi-automatic fashion seems like more risk than it's worth. And the time it'll take for CR and QA and the potential time it'll take to fix if something goes wrong makes this a net loss for me, I'm afraid.

Sorry for the run-around, @cclauss ; I thought we were on the same page!

**Pull request closed without merging**